test/object_store/test_basic.py

```diff
@@ -233,3 +234,20 @@ def list_bucket(s3_server):
         for o in objects:
             for ent in sstable_entries:
                 assert not o.startswith(str(ent[1])), f'Sstable object not cleaned, found {o}'
+
+@pytest.mark.asyncio
+async def test_misconfigured_storage(test_tempdir, s3_server, ssl):
+    '''creating keyspace with unknown endpoint is not allowed'''
+    # scylladb/scylladb#15074
+    with managed_cluster(test_tempdir, ssl, s3_server) as cluster:
+        print(f'Create keyspace (minio listening at {s3_server.address})')
+        replication_opts = format_tuples({'class': 'NetworkTopologyStrategy',
+                                          'replication_factor': '1'})
+        storage_opts = format_tuples(type='S3',
+                                     endpoint='unknown_endpoint',
+                                     bucket=s3_server.bucket_name)
```

Comment:
Why do you need the s3_server fixture, or its bucket_name, for this test? I think you don't need it?

Reply:
i concur with you. but i think Pavel might want to create a more real-world use case, a cluster is configured with some (correct) backend storage endpoints. but the user somehow tries to create a keyspace with a wrong endpoint.

Reply:
Well... I need a bucket name. It can be random as well, but strictly speaking the check is for misconfigured endpoint, not endpoint:bucket pair

Reply:
@tchaikov imagine that you are testing the "curl" tool and want to check what happens when you give it a malformed URL. Would you begin the test by running a real Web server on AWS and then running a curl command with a URL that has nothing to do with that server.

@xemul yes, you need a bucket name, but how is the bucket name from the real S3 server which you don't use relevant, more than the bucket name "dog"?

I admit it's not a disaster, but I just found it weird.

By the way, you keep saying that this is an "unconfigured" endpoint, but in fact, in this test you did configure an endpoint, you just set it to 'unknown_endpoint', which presumably doesn't exist in the DNS (but who knows, might ;-)). Shouldn't you also try not setting "endpoint" at all instead of setting it to garbage?

Another by the way - what happens if "endpoint" is it to an address that looks fine, e.g., "www.google.com", but it's just not an S3 server? Is this tested on creation time? I assume the answer is no?

Reply:
> By the way, you keep saying that this is an "unconfigured" endpoint, but in fact, in this test you did configure an endpoint, you just set it to 'unknown_endpoint', which presumably doesn't exist in the DNS (but who knows, might ;-)). Shouldn't you also try not setting "endpoint" at all instead of setting it to garbage?

Not setting endpoint is not allowed and there's test for that in cql-pytest/test_keyspace.py

```
# Test that not passing required parameters fails
def test_storage_options_required_param(cql, scylla_only):
    ksdef = "WITH REPLICATION = { 'class' : 'NetworkTopologyStrategy', 'replication_factor' : '1' } " \
            "AND STORAGE = { 'type' : 'S3', 'bucket' : '42' }"
    with pytest.raises(InvalidRequest):
        with new_test_keyspace(cql, ksdef):
            pass
```

> Another by the way - what happens if "endpoint" is it to an address that looks fine, e.g., "www.google.com", but it's just not an S3 server? Is this tested on creation time? I assume the answer is no?

Then it will create a client for that endpoint later and will get some error. Most likely this will first happen memtable flush time, for that case there's #15635

Reply:
> @tchaikov imagine that you are testing the "curl" tool and want to check what happens when you give it a malformed URL. Would you begin the test by running a real Web server on AWS and then running a curl command with a URL that has nothing to do with that server.
> 

ahh! good point. i am convinced. thanks.